index.js

```diff
@@ -10,25 +10,23 @@ function generate(argIndex) {
 		var promise = thunk(value);
 		thunk.then = promise.then.bind(promise);
 		thunk.catch = promise.catch.bind(promise);
-		return thunk;
 
-		function thunk(result) {
-			if (promise) {
-				// Prevent unhandled rejection errors if promise is never used, and only used as a thunk
-				promise.catch(function () {});
-			}
+		// Prevent unhandled rejection errors if promise is never used, and only used as a thunk
+		promise.catch(function () {});
```

Comment:
You can't do it this way.

We only want to suppress the `unhandledRejection` on the returned promise if it gets used as a `thunk`. The way you are doing it here, it suppresses it regardless.

We probably should write better tests